exporter/awsemfexporter/metric_translator.go

```diff
@@ -118,6 +134,278 @@ func (dps DoubleHistogramDataPointSlice) At(i int) DataPoint {
+			} else {
+				instrumentationLibName = ilm.InstrumentationLibrary().Name()
+			}
+
+			metrics := ilm.Metrics()
+			if metrics.Len() > 0 {
+				for k := 0; k < metrics.Len(); k++ {
+					metric := metrics.At(k)
+					if metric.IsNil() {
+						totalDroppedMetrics++
+						continue
+					}					
+					getGroupedMetrics(&metric, namespace, instrumentationLibName, groupedMetricMap, config)
+				}
+			}
+		}
+	}
+	return groupedMetricMap, totalDroppedMetrics
+}
+
+
+func getGroupedMetrics(metric *pdata.Metric, namespace string, instrumentationLibName string, groupedMetricMap map[string]*GroupedMetric, config *Config) (groupedMetrics []*GroupedMetric){
+	if metric == nil {
+		return
+	}
+
+	// Retrieve Datapoints
+	var dps DataPoints
+	switch metric.DataType() {
+		case pdata.MetricDataTypeIntGauge:
+			dps = IntDataPointSlice{metric.IntGauge().DataPoints()}
+		case pdata.MetricDataTypeDoubleGauge:
+			dps = DoubleDataPointSlice{metric.DoubleGauge().DataPoints()}
+		case pdata.MetricDataTypeIntSum:
+			dps = IntDataPointSlice{metric.IntSum().DataPoints()}
+		case pdata.MetricDataTypeDoubleSum:
+			dps = DoubleDataPointSlice{metric.DoubleSum().DataPoints()}
+		case pdata.MetricDataTypeDoubleHistogram:
+			dps = DoubleHistogramDataPointSlice{metric.DoubleHistogram().DataPoints()}
+		default:
+			config.logger.Warn("Unhandled metric data type.",
+				zap.String("DataType", metric.DataType().String()),
+				zap.String("Name", metric.Name()),
+				zap.String("Unit", metric.Unit()),)
+			return
+	}
+
+	if dps.Len() == 0 {
+		return
+	}
+
+	for m := 0; m < dps.Len(); m++ {
+		dp := dps.At(m)
+		if dp.IsNil() {
+			continue
+		}
+
+		fields := make(map[string]interface{})
+		s := []string{nameSpace, namespace, OTellibDimensionKey, instrumentationLibName}
+		fields[nameSpace] = namespace
+		fields[OTellibDimensionKey] = instrumentationLibName
+
+		dp.LabelsMap().ForEach(func(k string, v string) {
+			fields[k] = v
+			if (v != noInstrumentationLibraryName) {
+				s =append(s, k, v)
+			}
+		})
+
+		sort.Strings(s)
+		key := strings.Join(s, "")
+
+		if _, ok := groupedMetricMap[key]; ok {
+			updateGroupedMetric(dp, metric, key, groupedMetricMap)
+			groupedMetrics = append(groupedMetrics, groupedMetricMap[key])
+		} else {
+			groupedMetric := buildGroupedMetric(dp, fields, metric, config.DimensionRollupOption)
+			if groupedMetric != nil {
+				groupedMetricMap[key] = groupedMetric
+				groupedMetrics = append(groupedMetrics,groupedMetric)
+			}
+		}
+	}
+	return
+}
+
+// buildGroupedMetric builds GroupedMetric from Datapoint and pdata.Metric
+func buildGroupedMetric (dp DataPoint, fields map[string]interface{}, pMetricData *pdata.Metric, dimensionRollupOption string) (*GroupedMetric) {
+	var namespace string
+	timestamp := time.Now().UnixNano() / int64(time.Millisecond)
+	labels := make(map[string]string)
+	metrics := make(map[string]MetricInfo)
+	
+	namespace = fields[nameSpace].(string)
+
+	// Extract labels
+	for k, v := range fields {
```

Comment:
Let's build the labels up above in `getGroupedMetrics` like how it's done currently in `getCWMetrics`. This way, we don't have to perform type case to `string` which might cause some unintentional bugs. Alternatively, pass in `labels map[string]string` to this function and create `fields` from that.

Reply:
Actually, it looks like you're not even using `fields`? Can we just remove it then?